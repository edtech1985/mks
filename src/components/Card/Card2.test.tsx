import React from "react";
import { Card } from "./index";
import { render, screen } from "@testing-library/react";
import "jest-styled-components";

import "@testing-library/jest-dom/extend-expect";

describe("Card", () => {
  it("should render the card button", () => {
    const card = {
      key: 1,
      id: 1,
      name: "Product 1",
      brand: "Brand",
      photo: "Photo",
      description: "Description",
      price: "10",
      quantity: 1,
    };

    render(
      <Card
        key={card.key}
        id={card.id}
        name={card.name}
        brand={card.brand}
        description={card.description}
        photo={card.photo}
        price={card.price}
        quantity={card.quantity}
      />
    );
    screen.getByText("Brand Product 1");
  });
});

test("it works", () => {
  const card = {
    key: 1,
    id: 1,
    name: "Product 1",
    brand: "Brand",
    photo: "Photo",
    description: "Description",
    price: "10",
    quantity: 1,
  };

  const { container } = render(
    <Card
      key={card.key}
      id={card.id}
      name={card.name}
      brand={card.brand}
      description={card.description}
      photo={card.photo}
      price={card.price}
      quantity={card.quantity}
    />
  );
  expect(container).toMatchSnapshot();
});
