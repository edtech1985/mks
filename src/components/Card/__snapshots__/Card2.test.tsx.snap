// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c0 {
  width: 218px;
  height: 285px;
  border-radius: 8px;
  background: #fff;
  box-shadow: 0px 2px 8px 0px rgba(0, 0, 0, 0.14);
}

.c1 {
  width: 100%;
  height: auto;
  border-radius: 8px;
  align-items: center;
  display: flex;
  justify-content: center;
}

.c2 {
  width: auto;
  max-width: 170px;
  height: 138px;
  padding-top: 10px;
}

.c3 {
  width: 100%;
  height: 38px;
  padding: 13px 13px 8px;
  display: flex;
}

.c3 h2 {
  color: var(--text-gray);
  white-space: break-spaces;
  width: 60%;
  font-size: 16px;
  font-style: normal;
  font-weight: 400;
  line-height: 19px;
}

.c6 {
  height: 45px;
  padding: 0px 13px 12px;
  overflow: hidden;
}

.c6 p {
  color: var(--text-gray);
  height: auto;
  font-size: 10px;
  font-style: normal;
  font-weight: 300;
  line-height: 12px;
}

.c4 {
  display: flex;
  justify-content: center;
  vertical-align: middle;
  justify-content: space-between;
  align-items: center;
  align-content: center;
  align-self: center;
}

.c5 {
  width: 64px;
  height: 26px;
  border-radius: 5px;
  background: var(--bg-dark-gray);
  color: #fff;
  font-size: 15px;
  font-style: normal;
  font-weight: 700;
  text-align: center;
  align-items: center;
  justify-content: center;
  display: flex;
}

.c7 {
  width: 218px;
  height: 32px;
  border-radius: 0px 0px 8px 8px;
  background: var(--bg-blue);
  color: #fff;
  bottom: 0;
  border: none;
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 14px;
}

.c7 p {
  color: #fff;
  font-family: Montserrat;
  font-size: 14px;
  font-style: normal;
  font-weight: 600;
  line-height: 18px;
}

<div>
  <div
    class="c0"
    data-testid="card-container"
  >
    <div
      class="c1"
    >
      <img
        class="c2"
        data-testid="img"
        src="Photo"
      />
    </div>
    <div
      class="c3"
    >
      <h2>
        Brand
         
        Product 1
      </h2>
      <div
        class="c4"
      >
        <p
          class="c5"
          data-testid="price"
        >
          R$ 
          10
        </p>
      </div>
    </div>
    <div
      class="c6"
    >
      <p>
        Description
      </p>
    </div>
    <button
      class="c7"
    >
      <svg
        fill="none"
        height="16"
        viewBox="0 0 14 16"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3 1L1 3.7V13.15C1 13.8956 1.59695 14.5 2.33333 14.5H11.6667C12.403 14.5 13 13.8956 13 13.15V3.7L11 1H3Z"
          fill-rule="evenodd"
          opacity="0.737212"
          stroke="white"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
        <path
          d="M1 4.375H13"
          opacity="0.737212"
          stroke="white"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
        <path
          d="M10 7C10 8.24264 8.82475 9.25 7.375 9.25C5.92525 9.25 4.75 8.24264 4.75 7"
          opacity="0.737212"
          stroke="white"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
      <p>
        COMPRAR
      </p>
    </button>
  </div>
</div>
`;
